Add `norm` API for vectors

T45K opened the pull request:
## What
added `norm` APIs for vector type (i.e., `MultiArray<Float, D1>` and `MultiArray<Double, D1>`).
in this implementation, reusing existing `norm` methods by using origin vector (`mk.zeros(size)`).

## Why
currently, there are `norm` methods accepting only matrix (i.e., `MultiArray<Number, D2>`).
however, NumPy provides `norm` methods for vector.
sample: https://stackoverflow.com/a/43043160

i think it is better to provide the similar API for usability.

## Questions
- in this PR, i added new methods to `norm.kt`, but should i add them to `LinAlgEx.kt` file?
- where should i add tests?

T45K committed:
add norm API to Vector

devcrocod commented:
Thank you very much for the changes.

> In this PR, I added new methods to norm.kt, but should I add them to the LinAlgEx.kt file?

No, since you are reusing existing implementations, you don't need to add them to LinAlgEx.kt. That file contains methods whose implementations reside in other modules.

> Where should I add tests?

Since norm has two implementations, you need to test them in both modules:

In the Kotlin implementation: [KELinAlgTest.kt](https://github.com/Kotlin/multik/blob/develop/multik-kotlin/src/commonTest/kotlin/org/jetbrains/kotlinx/multik_kotlin/linAlg/KELinAlgTest.kt)
In the native (OpenBLAS) implementation: [NativeLinAlgTest.kt](https://github.com/Kotlin/multik/blob/develop/multik-openblas/src/commonTest/kotlin/org/jetbrains/kotlinx/multik/openblas/linalg/NativeLinAlgTest.kt)
You don't need to write complex tests. You can use the test from the native implementation as a reference: [NativeLinAlgTest.kt, Line 318](https://github.com/Kotlin/multik/blob/79dedd1cba815a4c1aa3d34c51646adc4abbd91b/multik-openblas/src/commonTest/kotlin/org/jetbrains/kotlinx/multik/openblas/linalg/NativeLinAlgTest.kt#L318).

T45K committed:
add tests

T45K commented:
@devcrocod 
thank you for advice!
i added tests to both of two test classes.

however, unfortunately, i don't have `OpenBLAS` env on my local.
so, may i ask you to execute tests i added to `NativeLinAlgTest` instead of me?

T45K committed:
revert import order

devcrocod commented:
All tests have passed.
Thank you very much for your contribution.

I can't say exactly when I will be able to release a new artifact with these changes, as I currently need to completely change the build infrastructure. This is further complicated by issues with the latest versions of Xcode.
